noir-projects/aztec-nr/aztec/src/contract_self.nr

```diff
@@ -198,6 +207,270 @@ impl<Storage, CallSelf, EnqueueSelf, CallSelfStatic, EnqueueSelfStatic, CallInte
     {
         emit_event_in_private(event, self.context, recipient, delivery_mode);
     }
+
```

Comment:
The docs here are mostly a copy-paster from `call_interfaces.nr`. I just:
- updated the arguments as they are different here,
- added the Example section which I thought is important because devs might not know how tho get the CallInterface arg.